src/dbos.ts

```diff
@@ -908,7 +908,9 @@ export class DBOS {
           timerFuncID,
         );
         if (!rres) return null;
-        if (rres?.cancelled) throw new DBOSTargetWorkflowCancelledError(workflowID); // TODO: Make semantically meaningful
```

Comment:
We're still not semantically meaningful :/

The main issue is that we run this logic on retrieve handlers, which are returned both when the workflow is a child, but also when the workflow is enqueued.

When the workflow is enqueued, where as the real message is "your code which enqueued and awaited workflow X got an exception because X timed out", the actual message will be "Workflow X awaited a workflow which was cancelled"


I think the solution is to have a different type of handler for Queues.

Reply:
If we change the awaited error to be like Python, then this is fine. If you enqueue something, await it, and it times out, you'll get the error "awaited workflow ID was cancelled", which is correct.